modules/gis/src/lib/binary-to-geojson.ts

```diff
@@ -1,4 +1,23 @@
-export function binaryToGeoJson(data, type, format) {
+import {GeoJSON, Feature, Geometry, Position, GeoJsonProperties} from 'geojson';
+import {Point, MultiPoint, LineString, MultiLineString, Polygon, MultiPolygon} from 'geojson';
+
+import {BinaryGeometryData, BinaryFeaturesData, BinaryGeometryType} from '../types';
+
+// We do not handle GeometryCollection, define a limited Geometry type that always has coordinates.
+type FeatureGeometry = Point | MultiPoint | LineString | MultiLineString | Polygon | MultiPolygon;
+
+/**
+ * Convert binary geometry representation to GeoJSON
+ * @param data   geometry data in binary representation
+ * @param type   Input data type: Point, LineString, or Polygon
+ * @param format Output format, either geometry or feature
+ * @return GeoJSON objects
+ */
+export function binaryToGeoJson(
+  data: BinaryGeometryData | BinaryFeaturesData,
+  type?: BinaryGeometryType,
+  format?: 'geometry' | 'feature'
+): GeoJSON | Feature[] | null {
```

Comment:
`@types/geojson` supports generics, so you should be able to use `Feature<FeatureGeometry>[]` both here and presumably throughout the file.